src/host.rs

```diff
@@ -55,10 +57,20 @@ impl Host {
         } else {
             let decoded = percent_decode(input.as_bytes());
             let domain = String::from_utf8_lossy(&decoded);
-            // TODO: Remove this check and use IDNA "domain to ASCII"
+
+            let mut domain = match idna::domain_to_ascii(&domain) {
+                Ok(s) => s,
+                Err(_) => return Err(ParseError::InvalidDomainCharacter)
+            };
+
             if !domain.is_ascii() {
-                Err(ParseError::NonAsciiDomainsNotSupportedYet)
-            } else if domain.find(&[
+                domain = match punycode::encode_str(&domain) {
+                    Some(dom) => "xn--".to_string() + &dom,
```

Comment:
I think that adding the `xn--` prefix should also be part of `domain_to_ascii`, but UTS#46 has a bug: https://github.com/whatwg/url/issues/53